### PR TITLE
Configurable AORC-CAMELS processing

### PR DESCRIPTION
**Pull request by glitt13**
These updates primarily create a user-configurable approach to processing AORC forcing data across CAMELS basins using via a yaml config file. Refer to generate.py for the basic documentation.

### CODE REVIEW

**Comment by glitt13** [comment]
@jmframe FYI - not sure what the code review process looks like but it would be good to see if this works on your end.

**Comment by jmframe** [comment]
@hellkite500 some of your files are changed here, is this all compatible?

File: forcing_prep/generate.py [hunk 1]
```diff
@@ -23,56 +37,53 @@
...
 
 
-def process_geo_data(gdf, data, name):
+def process_geo_data(gdf, data, name, y_lat_dim, x_lon_dim,out_dir = '', redo = False, cvar = 8, ctime_max = 120, cid = -1):
     print("Slicing data to domain")
     # Only need to load the raster for the geo data extent
     extent = gdf.total_bounds
     lats = slice(extent[1], extent[3])
     lons = slice(extent[0], extent[2])
-    # In  case the data is upside down, flip the y axis
```
Comment (hellkite500) [inline forcing_prep/generate.py]: This check should be left in, I think.  The AORC rasters are typically "upside down" and if it isn't flipped here before selecting the subset, you may not get correct extent.

File: forcing_prep/generate.py [hunk 2]
```diff
@@ -99,47 +110,81 @@ def process_geo_data(gdf, data, name):
...
+        path = Path(f"{out_dir}/camels_{uniq_name}")
         Path.mkdir(path, exist_ok=True)
+        # Write timeseries for each sub-catchment within CAMELS basin
         for name, data in cats:
-            data.to_csv(path / f"{name}.csv")
+            data.to_csv(path / f"{name}_{uniq_name}.csv")
+        # Write aggregated basin timeseries (all subcatchments averaged together)
         agg = df.groupby("time").mean()
-        agg.to_csv(path / f"camels_{b}_agg.csv")
+        agg.to_csv(path / f"camels_{uniq_name}_agg.csv")
```
Comment (hellkite500) [inline forcing_prep/generate.py:196]: add a newline at the end of file

**Review by hellkite500 — CHANGES_REQUESTED** [review]

File: forcing_prep/generate.py (hunk 1 above)
Comment (glitt13) [reply forcing_prep/generate.py]: I'll need to find where I was having issues with the flipping. In some cases, it caused one of the coordinates (latitude?) to have a dimension of size 0.

File: forcing_prep/generate.py (hunk 2 above)
Comment (glitt13) [reply forcing_prep/generate.py:196]: Good catch!

File: forcing_prep/generate.py (hunk 1 above)
Comment (hellkite500) [reply forcing_prep/generate.py]: This should be fixed in #4 addressing a bug in the original logic.

**Pull request merged**